ui/app/src/components/HistoryDialog/HistoryDialogContainer.tsx

```diff
@@ -57,12 +57,14 @@ import DialogFooter from '../DialogFooter/DialogFooter';
 import { HistoryDialogPagination } from './HistoryDialogPagination';
 import { historyStyles } from './HistoryDialog';
 import useSelection from '../../hooks/useSelection';
+import useFetchSandboxItems from '../../hooks/useFetchSandboxItems';
 
 export function HistoryDialogContainer(props: HistoryDialogContainerProps) {
   const { versionsBranch } = props;
   const { count, page, limit, current, rootPath, isConfig } = versionsBranch;
+  useFetchSandboxItems([versionsBranch.item.path]);
   // TODO: It'd be best for the dialog to directly receive a live item. Must change versions branch to only hold the path.
-  const item = useSelection((state) => state.content.itemsByPath[versionsBranch.item.path]) ?? versionsBranch.item;
```

Comment:
in this case the parsedLegacyItem wasn't getting to the content state, it was being used directly in the component. The item was being retrieved from the state and if it wasn't there it was using the one from the versions service. So that's what I did, using useFetchSandboxItems to retrieve and push to state, then use only from state